### ISSUE TITLE
Region-wise operation on ADE20k dataset

### ISSUE THREAD

**Issue by nmhkahn** [ZPdesu/SEAN#15]
Hello, thanks for the great work.

I just wonder how did you implement region-wise pooling and SEAN block on ADE20k dataset.
In CelebA dataset, it might manageable via for-loop since the number of styles is 19 whereas not sure it is possible to the ADE20k since it has more than 100 styles (or classes).
So could you share some experiences about the implementation or training/testing time in this dataset?

**Comment by ZPdesu** [comment]
That's a very good question.  At the code level, it's possible to use the torch.nn.ModuleList to manage your per-region Conv.  For simplicity, you can try to relax the constraint of per-region Conv and replace it with general 3*3 Convs

**Comment by NguyenTriTrinh** [comment]
Sorry to reopen this issue, but i am a little confused with this paragraph of codes,  is that means each class in the semantic label map needs a linear layer? So, are there 150 linear layers in the ADE20k dataset??? 
 https://github.com/ZPdesu/SEAN/blob/04c7536ff3fecd2d1a09c9ae046a1144636033a5/models/networks/normalization.py#L209
